File: src/main/java/Sigmath/engineering/Civil.java
package sigmath;



public class Civil
{
    private static double calculateBeamDeflection(double length, double load, double elasticity, double momentOfInertia)
    {
        return ((load * Basic.power(length, 3)) / (3 * elasticity * momentOfInertia));
    }

    private static double calculateStress(double force, double area)
    {
        return (force / area);
    }

    private static double calculateStrain(double initialLength, double finalLength, double initialDiameter)
    {
        return ((finalLength - initialLength) / (initialLength * (initialDiameter / 2)));
    }

    private static double calculateYoungsModulus(double stress, double strain)
    {
        return (stress / strain);
    }
}

File: Sigmath.ConvertedToRuby/engineering/civil.rb
module Sigmath
	class Civil
		def Civil.CalculateBeamDeflection(length, load, elasticity, momentOfInertia)
			return ((load * Basic.Power(length, 3)) / (3 * elasticity * momentOfInertia))
		end

		def Civil.CalculateStress(force, area)
			return (force / area)
		end

		def Civil.CalculateStrain(initialLength, finalLength, initialDiameter)
			return ((finalLength - initialLength) / (initialLength * (initialDiameter / 2)))
		end

		def Civil.CalculateYoungsModulus(stress, strain)
			return (stress / strain)
		end
	end
end